lib/tictactoe.rb

```diff
@@ -18,25 +19,45 @@ def render_board
 
   def place_token(position)
     board[position] = current_token
+    switch_turn
     render_board
   end
 
   def current_player
-    @current_token
+    current_token
   end
 
   def begin_player_turn
+    select_another_position_prompt = 'Select another position to place your token [1 - 9]: '
     desired_position = io.gets.to_i
     io.puts 'Select a position to place your token [1 - 9]: '
 
-    if board[desired_position] == ' '
+    if ![*1..9].include? desired_position
+      io.puts 'Invalid position'
+      io.puts select_another_position_prompt
+      render_board
+    elsif board[desired_position] == ' '
+      @move_valid = true
       place_token(desired_position)
     else
       io.puts 'That position is taken!'
+      io.puts select_another_position_prompt
+      render_board
      end
   end
 
   private
 
-  attr_accessor :board, :current_token, :io
+  attr_accessor :board, :current_token, :io, :move_valid
+
+  def switch_turn
+    if @move_valid
```

Comment:
Can we move this somewhere else? In my opinion, switch turn should do exactly what it entails. it shouldn't have to be concerned with if a move is valid or not.

Reply:
This is the same issue as the convo above. @move_valid doesn't exist anymore. I'll ask in the slack how to update the code shown here.

Reply:
That part of the code now looks like this: 
```
  attr_accessor :board, :current_token, :io

  def switch_turn
    if @current_token == 'X'
      @current_token = 'O'
    elsif @current_token == 'O'
      @current_token = 'X'
    end
  end
```

Reply:
👍